sycl/doc/extensions/proposed/SYCL_EXT_ONEAPI_GROUP_STATIC_ARRAY.asciidoc

```diff
@@ -0,0 +1,590 @@
+|===
+|Member function|Description
+
+|`template<std::size_t ItemsPerWorkItem = 1, std::int32_t dimensions = 1>
+static std::size_t memory_required(sycl::memory_scope scope, sycl::range<dimensions> local_range)`
+|Returns size of temporary memory (in bytes) that is required by the radix
+sorting algorithm defined by the sorter calling by `sort_over_group`.
+`ItemsPerWorkItem` is a parameter for `sycl::span<T, ItemsPerWorkItem>`
+that is an input parameter for `sort_over_group`, where `T` is a first template argument
+for `radix_sorter`. The function can be used for `sort_over_group` without `sycl::span`
+as an input parameter if `ItemsPerWorkItem == 1`.
+If `scope = sycl::memory_scope::work_group`,
+`local_range` is a local range of `sycl::nd_range` that was used to run the kernel;
+if `scope = sycl::memory_scope::sub_group`, `local_range` is a sub-group size.
+If other `scope` values are passed, behavior is unspecified.
+
+|`template<typename U, std::size_t ItemsPerWorkItem, std::int32_t dimensions = 1>
+static constexpr size_t
+key_value_memory_required(sycl::memory_scope scope, sycl::range<dimensions> r);`
+|Returns size of temporary memory (in bytes) that is required by the radix key-value
+sorting algorithm defined by the sorter calling by `sort_over_group`
+with `sycl::span<T, ItemsPerWorkItem>` and `sycl::span<U, ItemsPerWorkItem>`
+as input parameters, where `T` is a first template argument for `radix_sorter`.
+If `scope = sycl::memory_scope::work_group`,
+`local_range` is a local range of `sycl::nd_range` that was used to run the kernel;
+if `scope = sycl::memory_scope::sub_group`, `local_range` is a sub-group size.
+If other `scope` values are passed, behavior is unspecified.
+
+|===
+
+=== SYCL Properties for Interfaces with Static Private Arrays
+
+Consider the `sort_over_group` function (following can be applied for
+`exclusive_scan_over_group` and `inclusive_scan_over_group` as well).
+
+Let `N` is a work-group size, `ItemsPerWorkItem` is a number of elements
+that are processed by one work-item.
+
+Sorting is performed across `N * ItemsPerWorkItem` elements in the group.
+Let `r` is a virtual range for sorting of `N * ItemsPerWorkItem` elements.
+Placing of the result data to `ItemsPerWorkItem` elements per each work-item
+could be done in 2 layouts:
+
+a) Data from the `[r + id * ItemsPerWorkItem; r + (id + 1) * ItemsPerWorkItem)` virtual range
+placed into the private memory under the span for `id`-th work-item.
+
+b) `i * N + id` element of `r` fill the `i`-th element of the private memory
+under the span for `id`-th work-item.
+
+To specify a correct layout for placing of resulting data there are 2 properties
+that satisfy SYCL Properties Extension requirements:
+
+1.`sycl::ext::oneapi::experimental::property::close` to specify layout described in a).
+
+2.`sycl::ext::oneapi::experimental::property::spread` to specify layout described in b).
```

Comment:
Question about naming is difficult. I have this as an open question number 6 below in the document. We need to think more about it

Reply:
Looking at this again after the discussion in https://github.com/intel/llvm/pull/5571#discussion_r805990027, I don't think you can use these names directly anyway.  If the name of the property is changed to `layout` (or some variation of that), then the property will need to change to something like `layout<close>` or `layout<spread>`.

I think "close" and "spread" are even more confusing when tied to the word "layout", because the phrase "close layout" doesn't have an obvious interpretation.